### ISSUE TITLE
Feature request: abbreviate package names in Java Projects view

### ISSUE THREAD

**Issue by jjoslet** [microsoft/vscode-java-pack#1211]
## Suggestion

In the **Java Projects** view, allow to visually abbreviate package names.

The idea comes from this Eclipse IDE feature:
![Eclipse Example](https://github.com/microsoft/vscode-java-pack/assets/18058774/b9137514-fe19-4a9b-a5be-d6e71299f006)

Defines abbreviate rules:
```
com.my.company={CP}
com.my.company.myprojecta={A}
com.my.company.myprojectb={B}
```
These rules are then used to abbreviate package names in the view.

## Examples

* `com.my.company.myprojecta.services` will be displayed `{A}.services`
* `com.my.company.myprojecta.services.remote` will be displayed `{A}.services.remote`
* `com.my.company.myprojectb.services.postal` will be displayed `{B}.services.postal`
* `com.my.company.myprojectc.services.git` will be displayed `{CP}.myprojectc.services.git`

**Comment by nickzhums** [comment]
@jjoslet thanks for raising this feature request. Just a quick question - How heavily are you using the Java Project View and are you switching between File Explorer and Java Project View regularly?

**Comment by jjoslet** [comment]
I mainly use Java Project View cause I personally prefer the flat package view which seems not possible on the File Explorer.

I switch to the File Explorer when an action is not possible on the Java Project view, for example: multiple element selection and moving some files to other folder.

**Comment by jdneo** [comment]
Ah! This feature looks very useful! Thanks for the feedback.

BTW file drag and drop should be supported in Java Project explorer, let us know if there is anything missing.

**Comment by jjoslet** [comment]
@nickzhums yes, I find this very useful. I regularly have extra resource files in a Java project and I appreciate to manage all my project's files in the same view. Example of such resources:
* SQL DDL files
* SQL Schema update files
* Ascidoctor documentation under `src/docs/asciidoc` as recommended by the official Asciidoctor Maven Plugin

These resources are not in the `src/main/resources` folder since I don't need them in the final jar.

@jdneo Maybe it'll need another issue, here is the scenario; I create a file `.env` at the root of a Maven project (near the `pom.xml`) and then I try to move it to different folder:
* the `.vscode` folder that contains a `launch.json` -> doesn't work
* a newly created folder `my-folder` -> doesn't work
* the `src\main\resources\` folder -> work as expected
After moving it to `src\main\resources\` folder, I try to move it back to the root, this doesn't work.

My real use case was to create a `.env` files with environment variables to configure my Spring Boot application launched by the `launch.json` file using `"envFile": "${workspaceFolder}/.vscode/.env"`.

Do you want me to create another issue for that ?
